core/chains/evm/gas/models.go

```diff
@@ -271,12 +271,25 @@ func (e *evmFeeEstimator) GetFee(ctx context.Context, calldata []byte, feeLimit
 		fee.DynamicTipCap = dynamicFee.TipCap
 		return
 	}
-
 	// get legacy fee
 	fee.Legacy, chainSpecificFeeLimit, err = e.EvmEstimator.GetLegacyGas(ctx, calldata, feeLimit, maxFeePrice, opts...)
 	if err != nil {
 		return
 	}
+
+	if e.geCfg.CostMax() != nil {
```

Comment:
You will need to also do this same logic inside the BumpFee().
Actually, 99% of times, that is the function where we hit this limit, not here.

You can create a single utility function, that takes the calculated Fee, and then adjusts it back to under the limit based on maxCost(), and let both GetFee() and BumpFee() call that utility function.

Reply:
done. PTAL